File: java/src/main/java/it/giannibombelli/refctor_challenge/FizzBuzz.java
package it.giannibombelli.refctor_challenge;

public class FizzBuzz implements FizzBuzzInterface {
    @Override
    public String say(int number) {
        if (number % 3 == 0 && number % 5 == 0) {
            return "FizzBuzz";
        }
        if (number % 3 == 0) {
            return "Fizz";
        }
        if (number % 5 == 0) {
            return "Buzz";
        }

        return String.valueOf(number);
    }
}

File: kotlin/src/main/kotlin/it/giannibombelli/refctor_challenge/FizzBuzz.kt
package it.giannibombelli.refctor_challenge

class FizzBuzz : FizzBuzzInterface {
    override fun say(number: Int): String {
        if (number % 3 == 0 && number % 5 == 0) {
            return "FizzBuzz"
        }
        if (number % 3 == 0) {
            return "Fizz"
        }
        if (number % 5 == 0) {
            return "Buzz"
        }

        return number.toString()
    }
}
